minor fix to docs about exit-script

ColemanTom opened the pull request:
I noticed that the `exit-script` section in the docs referred to itself rather than `err-script`, so this just adjusts that.

I did contemplate another adjustment. In the order of the docs, it has `exit-script` above `err-script`. But in the `See also ...` parts for these script sections, it lists `err-script` before `exit-script`. I'm not sure what the ordering in here is meant to be, but I would suggest that it should be order of execution really for both - it makes the documentation more intuitive.

hjoliver commented:
@ColemanTom - thanks, good spotting.

These sections (and the "See also" lists) are more or less un-ordered (probably due to organic evolution of the docs over time).  But you are right, they should be in the right order.

The right order is as shown in this diagram: https://github.com/cylc/cylc/issues/2962#issuecomment-472298847

(only one of exit-script or err-script gets run, depending on exit status)

Would you like to amend this PR with changes to order these correctly?

hjoliver commented:
(Also, when you're done, could you cherry-pick your changes to a new PR for the 7.8.x branch - thanks)

ColemanTom commented:
K. One clarification first though, https://github.com/cylc/cylc/pull/3066/files#diff-8838fd940264b809b57525f23c2137deR1370

There is a `.. _ScriptItem` just above the `script` section. I can't see that used anywhere, so is it ok for me to delete it, or is there a missing ref somewhere else in the documentation?

hjoliver commented:
I'm guessing that's some kind of typo (seems to be the only occurrence) but should maybe let @sadielbartholomew comment (I'm not too familiar with the new documentation format yet).

kinow commented:
>There is a .. _ScriptItem just above the script section. I can't see that used anywhere, so is it ok for me to delete it, or is there a missing ref somewhere else in the documentation?

Chiming in. I did a `grep` in the `docs` folder, and found no other references for that link. But then in `gcylc-config-ref.rst` there's also `_GcylcRCReference`, and greping for `grep -r -H -i gcylcrcreference` doesn't find anything 😕

Probably safe to remove IMO, if you need to.

ColemanTom commented:
I've updated the orders, and removed that `ScriptItem` line. I can easily add it back in if @sadielbartholomew wants me to.

I'll cherry-pick once this has been approved in case more changes are required..

hjoliver commented:
Thanks @ColemanTom 🎉

matthewrmshin commented:
Back port PR is #3070?
